P202-Training.Domain/IUsersService.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+using P202.Training.Domain.Models;
+using P202.Training.Data.Repositories;
+
+namespace P202.Training.Domain
+{
+    public interface IUsersService
+    {
+        void CreateUser(User users, IUserRepository userRepository);
```

Comment:
You shouldn't be sending the repository as a parameter in each method. That should be injected in the service constructor. And it definitely shouldn't be part of the interface